buda/entities/action_taken.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package buda;
+
+import "buda/entities/uuid.proto";
+import "google/protobuf/timestamp.proto";
+
+/**
+ * ActionTaken is an action that a user takes in an application directly
+ */
+message ActionTaken {
+    Uuid id = 1;
+    // The application where the action is performed
+    string application = 2;
+    // The location in the application UI where the action is performed
+    string location = 3;
+    // A unique label for the action performed
+    string action = 4;
```

Comment:
@djfarrelly great point about a convention on which tense to use. My hunch is past tense to stick with the ActionTaken tense? It's also a subtle hint that this is an immutable event